src/commands/getExtractCommand.ts

```diff
@@ -71,10 +72,34 @@ export const getExtractCommand = (context: vscode.ExtensionContext) => {
     });
 
     try {
-      await updateFiles(editor, selection, changes);
-    } catch (error: unknown) {
-      const { message } = error as Error;
-      vscode.window.showErrorMessage(message);
+      await vscode.window.withProgress(
+        {
+          location: vscode.ProgressLocation.Notification,
+          cancellable: false,
+          title: "Angular Extractor",
+        },
+        startProgress([
+          {
+            execute: generateNewComponent(
```

Comment:
I think it would be better to create the function without parameters here instead of in the angular implementations.
```ts
execute: () => generateNewComponent(
  useNpx,
  componentName,
  componentDirectory
),
```

Reply:
Yes it's definitely better. Like the idea and refactored it.